### DISCUSSION TITLE
How can I include the mode in the output folder ?

### DISCUSSION THREAD

**Discussion by stevebeauge** [wxt-dev/wxt#820] [Q&A] [answered]
Hello,

I migrated from plasmo to wxt and I'm very happy with the move!

Though I miss a small thing from plasmo, which the ability to include both browser and "mode" in the output folder

Our company is selling a browser extension that should be built for both edge and chrome (as of today, the list may expand over time) and we are distributing both the PROD version and the DEV version (for testers, ...).

The current behavior of wxt is to generate the result inside `.ouput/{browser}-{manifestversion}`.

In our release process, we can generate all combinations (edge/prod, edge/dev, chrome/prod, chrome/dev) at once.

Because the mode isn't part of the output folder, the prod and dev build are overlapping.

I would like to be able to emit the build inside `.ouput/{browser}-{manifestversion}-{mode}`.

Is there any way to do this?

Should it be possible to replace the `outdir? : string` option by `outdir? : string | (env: ConfigEnv) => string` (similar to the `manifest` option).

Or should it be possible to add cli arguments that may overwrite some parameters? `wxt zip -b msedge --outdir .output/msedge-mv3-prod` ?

FYI, I worked around the issue by scripting what I need. I have a build script that will generate all the combinations of supported browser / mode and call the WXT build or zip API. After each build, I can "move" the output to the desired location. It's working, but required a bit of plumbing.

Thanks!

**Comment by aiktb** [comment]
More than just command line arguments, defineConfig should support functions, like vite.config.ts:
```ts
// This could even be the async function
export default defineConfig(({ command, mode }) => {
  return {
    // vite config
  }
})
```

**Comment by aklinker1** [comment, marked as answer]
For now, you can change the output directory using a module:

```ts
// modules/out-dir.ts
export default defineWxtModule((wxt) => {
  wxt.hooks.hook('ready', (wxt) => {
    wxt.config.outDir += `-${WXT.config.mode}`
  });
});
```

**Reply by stevebeauge** [reply]
Works like a charm.
I wasn't aware of the module feature (confusion with es module).

Thx
